osbuild/util/runners.py

```diff
@@ -0,0 +1,108 @@
+import os.path
+import platform
+import shutil
+import subprocess
+import sys
+from contextlib import contextmanager
+
+
+def ldconfig(*dirs):
+    # ld.so.conf must exist, or `ldconfig` throws a warning
+    subprocess.run(["touch", "/etc/ld.so.conf"], check=True)
+
+    if len(dirs) > 0:
+        with open("/etc/ld.so.conf", "w", encoding="utf8") as f:
+            for d in dirs:
+                f.write(f"{d}\n")
+            f.flush()
+
+    subprocess.run(["ldconfig"], check=True)
+
+
+def sysusers():
+    try:
+        subprocess.run(
+            ["systemd-sysusers"],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=True,
+        )
+    except subprocess.CalledProcessError as error:
+        sys.stderr.write(error.stdout)
+        sys.exit(1)
+
+
+@contextmanager
+def create_machine_id_if_needed(tree=""):
+    """Create a machine-id with a fake machine id if it does not exist"""
+    path = f"{tree}/etc/machine-id"
+    path_exists = os.path.exists(path)
+    try:
+        if path_exists:
+            yield
+        os.makedirs(f"{tree}/etc", exist_ok=True)
+        with open(path, "w", encoding="utf8") as f:
+            # create a fake machine ID to improve reproducibility
+            f.write("ffffffffffffffffffffffffffffffff\n")
+            os.fchmod(f.fileno(), 0o400)
+            yield
```

Comment:
The above needs to be in the `else` condition for line 41, otherwise it still gets executed (just when the contextmanager exits) and the contextmanager will yield twice which will throw a `RuntimeError`.